src/features/runtime-checks.js

```diff
@@ -573,4 +577,204 @@ export default class RuntimeChecks extends ContentFeature {
+                    this.overloadStorageWithMemory(storageConfig, storageType)
+                } else if (storageConfig.scheme === 'session') {
+                    this.overloadStorageWithSession(storageConfig, storageType)
+                }
+            }
+        })
+        const breakpoints = this.getFeatureSetting('breakpoints')
+        const screenSize = { height: screen.height, width: screen.width };
+        ['innerHeight', 'innerWidth', 'outerHeight', 'outerWidth', 'Screen.prototype.height', 'Screen.prototype.width'].forEach(sizing => {
+            if (sizing in genericOverloads) {
+                const sizingConfig = genericOverloads[sizing]
+                this.overloadScreenSizes(sizingConfig, breakpoints, screenSize, sizing, sizingConfig.offset || 0)
+            }
+        })
+    }
+
+    overloadDate (config) {
+        const offset = (new Date()).getTimezoneOffset()
+        globalThis.Date = new Proxy(globalThis.Date, {
+            construct (target, args) {
+                const constructed = Reflect.construct(target, args)
+                if (getTaintFromScope(this, arguments, config.stackCheck)) {
+                    // Falible in that the page could brute force the offset to match. We should fix this.
+                    if (constructed.getTimezoneOffset() === offset) {
+                        return constructed.getUTCDate()
+                    }
+                }
+                return constructed
+            }
+        })
+    }
+
+    overloadDateGetTimezoneOffset (config) {
+        const offset = (new Date()).getTimezoneOffset()
+        defineProperty(globalThis.Date.prototype, 'getTimezoneOffset', {
+            configurable: true,
+            enumerable: true,
+            writable: true,
+            value () {
+                if (getTaintFromScope(this, arguments, config.stackCheck)) {
+                    return 0
+                }
+                return offset
+            }
+        })
+    }
+
+    overloadHighEntropyValues (config) {
+        if (!('NavigatorUAData' in globalThis)) {
+            return
+        }
+
+        const originalGetHighEntropyValues = globalThis.NavigatorUAData.prototype.getHighEntropyValues
+        defineProperty(globalThis.NavigatorUAData.prototype, 'getHighEntropyValues', {
+            configurable: true,
+            enumerable: true,
+            writable: true,
+            value (hints) {
+                let hintsOut = hints
+                if (getTaintFromScope(this, arguments, config.stackCheck)) {
+                    // If tainted override with default values (using empty array)
+                    hintsOut = []
+                }
+                return Reflect.apply(originalGetHighEntropyValues, this, [hintsOut])
+            }
+        })
+    }
+
+    overloadStorageWithMemory (config, key) {
+        class MemoryStorage {
+            #data = {}
+
+            setItem (id, val) {
+                this.#data[id] = String(val)
```

Comment:
how closely do we need to match APIs here? because `localStorage.setItem('foo')`  (missing the value) would throw in the browser, but this implementation will set the key with `undefined` as the value